spharpy/spherical.py

```diff
@@ -1115,3 +1118,385 @@ def sph_identity_matrix(n_max, type='n-nm'):
+            self._reset_compute_attributes()
+            self._basis_type = value
+
+    @property
+    def inverse_method(self):
+        """Get or set the type of inverse transform."""
+        return self._inverse_method
+
+    @inverse_method.setter
+    def inverse_method(self, value):
+        """Set the inverse transform type."""
+        allowed = ["pseudo_inverse", "quadrature", "auto"]
+        if value not in allowed:
+            raise ValueError("Invalid inverse method. Allowed values are 'inverse', "
+                             "'quadrature', or 'auto'.")
+        # If using simple pf.Coordinates and not SamplingSphere, 'auto' is not allowed
+        if not isinstance(self.coordinates, SamplingSphere) and value == "auto":
+            raise ValueError("inverse_method='auto' is not allowed without SamplingSphere coordinates. " \
+            "Please specify 'pseudo_inverse' or 'quadrature'.")
+        # If using quadrature, weights must be set
+        if value == "quadrature" and self.coordinates.weights is None:
+            raise ValueError("Weights must be set for quadrature inverse method")
+        if value != self._inverse_method:
+            self._reset_compute_attributes()
+            self._inverse_method = value
+
+    @property
+    def channel_convention(self):
+        """Get or set the channel ordering convention."""
+        return self._channel_convention
+
+    @channel_convention.setter
+    def channel_convention(self, value):
+        """Set the channel order convention."""
+        if value not in ["acn", "fuma"]:
+            raise ValueError("Invalid channel convention, "
+                             "currently only 'acn' "
+                             "and 'fuma' are supported"
+                             )
+        if value == "fuma" and self.n_max > 3:
+            raise ValueError("n_max > 3 is not allowed with 'fuma' " \
+                            "channel convention")
+        if value != self._channel_convention:
+            self._reset_compute_attributes()
+            self._channel_convention = value
+
+    @property
+    def normalization(self):
+        """Get or set the normalization convention."""
+        return self._normalization
+
+    @normalization.setter
+    def normalization(self, value):
+        """Set the normalization convention."""
+        if value not in ["n3d", "maxN", "sn3d"]:
+            raise ValueError(
+                "Invalid normalization, "
+                "currently only 'n3d', 'maxN', 'sn3d' are "
+                "supported"
+            )
+        if value == "maxN" and self.n_max > 3:
+            raise ValueError("n_max > 3 is not allowed with " \
+                            "'maxN' normalization")
+        if value != self._normalization:
+            self._reset_compute_attributes()
+            self._normalization = value
+
+    @property
+    def basis(self):
+        """Get or set the spherical harmonic basis matrix."""
```

Comment:
This does not have a setter right? In this case I would suggest
```suggestion
        """Get the spherical harmonic basis matrix."""
```